lib/internal/util/comparisons.js

```diff
@@ -17,18 +35,65 @@ const {
   ObjectPrototypeHasOwnProperty: hasOwn,
   ObjectPrototypePropertyIsEnumerable: hasEnumerable,
   ObjectPrototypeToString,
+  Promise,
+  RegExp,
   SafeSet,
+  Set,
+  String,
   StringPrototypeValueOf,
+  Symbol,
   SymbolPrototypeValueOf,
   TypedArrayPrototypeGetByteLength: getByteLength,
   TypedArrayPrototypeGetSymbolToStringTag,
+  Uint16Array,
+  Uint32Array,
   Uint8Array,
+  Uint8ClampedArray,
+  WeakMap,
+  WeakSet,
 } = primordials;
 
 const { compare } = internalBinding('buffer');
 const assert = require('internal/assert');
 const { isURL } = require('internal/url');
 const { isError } = require('internal/util');
+const { Buffer } = require('buffer');
+
+const wellKnownConstructors = new SafeSet();
+wellKnownConstructors.add(Array);
+wellKnownConstructors.add(ArrayBuffer);
+wellKnownConstructors.add(BigInt);
+wellKnownConstructors.add(BigInt64Array);
+wellKnownConstructors.add(BigUint64Array);
+wellKnownConstructors.add(Boolean);
+wellKnownConstructors.add(Buffer);
+wellKnownConstructors.add(DataView);
+wellKnownConstructors.add(Date);
+wellKnownConstructors.add(Error);
+if (Float16Array) { // TODO(BridgeAR): Remove when regularly supported
+  wellKnownConstructors.add(Float16Array);
+}
```

Comment:
We probably do not need the condition here, it should be harmless to add `undefined` to the set – I'm fine keeping it to be clear

Reply:
That would cause null prototypes to not properly being distinguishable anymore. I already added a test case to cover that.